tests/Integration/SpecialPersistentPageIdentifierResolverIntegrationTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PersistentPageIdentifiers\Tests\Integration;
+
+use ProfessionalWiki\PersistentPageIdentifiers\Adapters\DatabasePersistentPageIdentifiersRepo;
+use ProfessionalWiki\PersistentPageIdentifiers\Application\PersistentPageIdentifiersRepo;
+use ProfessionalWiki\PersistentPageIdentifiers\EntryPoints\SpecialPersistentPageIdentifierResolver;
+use ProfessionalWiki\PersistentPageIdentifiers\Tests\PersistentPageIdentifiersIntegrationTest;
+use Status;
+
+/**
+ * @covers \ProfessionalWiki\PersistentPageIdentifiers\EntryPoints\SpecialPersistentPageIdentifierResolver
+ * @group Database
+ */
+class SpecialPersistentPageIdentifierResolverIntegrationTest extends PersistentPageIdentifiersIntegrationTest {
+
+	private PersistentPageIdentifiersRepo $repo;
+
+	protected function setUp(): void {
+		parent::setUp();
+		$this->tablesUsed[] = 'persistent_page_ids';
+		$this->repo = new DatabasePersistentPageIdentifiersRepo( $this->db );
+	}
+
+	protected function newSpecialPage(): SpecialPersistentPageIdentifierResolver {
+		return new SpecialPersistentPageIdentifierResolver();
+	}
+
+	public function testExecuteWithValidIdRedirects(): void {
+		$page = $this->createPageWithText();
```

Comment:
For some reason, the pages created in this test are leaking to `GenerateMissingIdentifiersTest` only in the CI for `REL1_39` and `REL1_41`. I don't know how to address that properly.

@JeroenDeDauw what do you think?